src/main/java/com/library/controller/BookController.java

```diff
@@ -47,7 +54,9 @@ public List<BookDto> getAllBooks(@Valid @RequestParam Integer page, @Valid @Requ
      */
     @GetMapping(value = "/{id}")
     public BookDto getBookById(@Valid @PathVariable("id") Long bookId) {
-        return bookService.findById(bookId);
+        BookDto book = bookService.findById(bookId);
+        log.debug(LocalDateTime.now().format(DATE_TIME_FORMATTER).concat(" - ").concat(String.format("BookController.getBookById(%s) - ", bookId)).concat(book.toString()));
```

Comment:
1) What is the difference between String and Stringbuilder classes? What class it is "needed" to use when some string is constructed from several strings?

2) Why logback.xml configuration is used in app when it is possible to create detailed message by hand?

3) Just look at this message creating:
several concats, inside concat-method param string.format.... why not to use one string.format with several params {%s}?
It is ugly code too (read point 2) but it looks more readable and without nested parentheses.

Reply:
already deleted this part